api/types/types_config_integration_compat.go

```diff
@@ -21,16 +21,47 @@ const (
 
 	// ConfigOldIntegrationVolPathCache is a config key.
 	ConfigOldIntegrationVolPathCache = ConfigOldRoot + ".path.cache"
+
+	//ConfigOldDocker is a config key.
+	ConfigOldDocker = "docker"
+
+	//ConfigOldDockerFsType is a config key.
+	ConfigOldDockerFsType = ConfigOldDocker + ".fsType"
+
+	//ConfigOldDockerVolumeType is a  config key.
+	ConfigOldDockerVolumeType = ConfigOldDocker + ".volumeType"
+
+	//ConfigOldDockerIOPS is a config key.
+	ConfigOldDockerIOPS = ConfigOldDocker + ".iops"
+
+	//ConfigOldDockerSize is a config key.
+	ConfigOldDockerSize = ConfigOldDocker + ".size"
+
+	//ConfigOldDockerAvailabilityZone is a config key.
+	ConfigOldDockerAvailabilityZone = ConfigOldDocker + ".availabilityZone"
+
+	//ConfigOldDockerMountDirPath is a config key.
+	ConfigOldDockerMountDirPath = ConfigOldDocker + ".mountDirPath"
+
+	//ConfigOldDockerLinuxVolumeRootPath is a config key.
+	ConfigOldDockerLinuxVolumeRootPath = "linux.volume.rootpath"
 )
 
 // BackCompat ensures keys can be used from old configurations.
 func BackCompat(config gofig.Config) {
 	checks := [][]string{
-		{ConfigIntegrationVolMountPreempt, ConfigOldIntegrationVolMountPreempt},
-		{ConfigIntegrationVolCreateDisable, ConfigOldIntegrationVolCreateDisable},
-		{ConfigIntegrationVolRemoveDisable, ConfigOldIntegrationVolRemoveDisable},
-		{ConfigIntegrationVolUnmountIgnoreUsed, ConfigOldIntegrationVolUnmountIgnoreUsed},
-		{ConfigIntegrationVolPathCache, ConfigOldIntegrationVolPathCache},
+		{ConfigIgVolOpsMountPreempt, ConfigOldIntegrationVolMountPreempt},
+		{ConfigIgVolOpsCreateDisable, ConfigOldIntegrationVolCreateDisable},
+		{ConfigIgVolOpsRemoveDisable, ConfigOldIntegrationVolRemoveDisable},
+		{ConfigIgVolOpsUnmountIgnoreUsed, ConfigOldIntegrationVolUnmountIgnoreUsed},
+		{ConfigIgVolOpsPathCache, ConfigOldIntegrationVolPathCache},
+		{ConfigIgVolOpsCreateDefaultFsType, ConfigOldDockerFsType},
+		{ConfigIgVolOpsCreateDefaultType, ConfigOldDockerVolumeType},
+		{ConfigIgVolOpsCreateDefaultIOPS, ConfigOldDockerIOPS},
+		{ConfigIgVolOpsCreateDefaultSize, ConfigOldDockerSize},
+		{ConfigIgVolOpsCreateDefaultAZ, ConfigOldDockerAvailabilityZone},
+		{ConfigIgVolOpsMountPath, ConfigOldDockerMountDirPath},
+		{ConfigIgVolOpsMountRootPath, ConfigOldDockerLinuxVolumeRootPath},
```

Comment:
All said and done, either work, so I'll merge. I was k let ever curious as to why you elected to use the array.